### PR TITLE
Make sure type can be unregistered successfully

### PR DESCRIPTION
**Pull request by Barry-Xu-2018**

### CODE REVIEW

**Comment by Barry-Xu-2018** [comment]
This patch is related to https://github.com/ros2/rmw_fastrtps/issues/433#issuecomment-693976261.  
And fixing way is the same as #434

**Comment by Barry-Xu-2018** [comment]
@ivanpauno @fujitatomoya

The patch fix the same problem like #434.  
Please help to review

**Review by fujitatomoya — APPROVED** [review]
this makes sense to me.

**Comment by fujitatomoya** [comment]
CC: @MiguelCompany

File: rmw_fastrtps_dynamic_cpp/src/subscription.cpp [hunk 1]
```diff
@@ -177,6 +177,14 @@ create_subscription(
     RMW_SET_ERROR_MSG("create_subscriber() could not create subscriber");
     return nullptr;
   }
+  auto cleanup_subscription = rcpputils::make_scope_exit(
```
Comment (ivanpauno) [inline rmw_fastrtps_dynamic_cpp/src/subscription.cpp:180]: shouldn't lines 121-129 be deleted?
seems like a double free if not

**Comment by Barry-Xu-2018** [comment]
@ivanpauno 

Fixed. Please review again.

**Review by ivanpauno — APPROVED** [review]

**Comment by ivanpauno** [comment]
CI up to `test_rmw_implementation`, `rcl`:

* Linux [Build Status](http://ci.ros2.org/job/ci_linux/12424/)
* Linux-aarch64 [Build Status](http://ci.ros2.org/job/ci_linux-aarch64/7391/)
* macOS [Build Status](http://ci.ros2.org/job/ci_osx/10134/)
* Windows [Build Status](http://ci.ros2.org/job/ci_windows/12337/)

**Pull request merged**